notifications.bs

```diff
@@ -539,14 +555,48 @@ interpreted as a language tag. Validity or well-formedness are not enforced. [[!
 platform supports activation, the user agent must (unless otherwise specified) run these steps:
 
 <ol>
+ <li><p>Let <var>action</var> be null.
+
+ <li><p>If one of <var>notification</var>'s <a for=notification>actions</a> was activated by the end
+ user, then set <var>action</var> to that <a for=/>action</a>.
+
+ <li><p>Let <var>navigationURL</var> be <var>notification</var>'s
+ <a for=notification>navigation URL</a>.
+
+ <li>
+  <p>If <var>action</var> is non-null, then set <var>navigationURL</var> to <var>action</var>'s
+  <a for=action>navigation URL</a>.
+
+  <p class=note>This intentionally makes it so that when an <a for=/>action</a>'s
+  <a for=action>navigation URL</a> is null, it falls through to the <code>click</code> event,
+  providing more flexibility to the web developer.
+
+ <li>
+  <p>If <var>navigationURL</var> is non-null:
+
+  <ol>
+   <li><p>If there is an existing <a for=/>top-level traversable</a> <var>traversable</var> within
+   the user agent's <a for="user agent">top-level traversable set</a> whose
+   <a for="navigable">active document</a>'s <a for=Document>URL</a> <a for=url>equals</a>
+   <var>navigationURL</var> with <a for=url/equals><i>exclude fragments</i></a> set to true, then
```

Comment:
Wouldn't this always compare the full path? Should we have some base path to compare instead? Otherwise I think each click would effectively always open a new tab unless the user already has a tab for the exact same URL.

Having a base URL would be useful for Slack example, as the URL is `https://app.slack.com/client/(workspace)/(channel)`, and the notification would pass `https://app.slack.com/client/(workspace)` as the base url to only match the same workspace.

Reply:
I think Slack would probably opt to handle the notification in its service worker if there is one, using this as fallback. Perhaps over time we can address more use cases here, but having some deployment experience first would help a lot with that.